mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -207,6 +208,10 @@ private boolean actionProfileMatches(ProfileSnapshotWrapper actionProfileWrapper
             if (matchedId != null) {
               record.setMatchedId(matchedId);
               record.setGeneration(null); // in case the same record is re-imported, generation should be calculated on SRS side
+              String instanceId = getValue(record, "999", 'i');
+              if (instanceId != null) {
```

Comment:
can be empty?

Reply:
@OleksiiKuzminov Thank you for this comment!
Added check for both: matchedId(from previous logic) and instanceId.